devicetypes/vision-kuowei/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,266 @@
+			commands << "delay 300"			
+			commands << encap(0, zwave.configurationV1.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}		
+	response(commands + refresh())
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug("'$description' parsed to $result")
+	return createEvent(result)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, endpoint=null) {
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, endpoint=null) {
+	(endpoint == 1) ? [name: "switch", value: cmd.value ? "on" : "off"] : [:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	if (cmd.commandClass == 0x6C && cmd.parameter.size >= 4) { // Supervision encapsulated Message		
+		cmd.parameter = cmd.parameter.drop(2)
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x25: 1, 0x20: 1])
+	if (cmd.sourceEndPoint == 1) {
+		zwaveEvent(encapsulatedCommand, 1)
+	} else { // sourceEndPoint == 2
+		childDevices[0]?.handleZWave(encapsulatedCommand)
+		[:]
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	if (endpoint == null) log.debug("$device.displayName: $cmd")
+	else log.debug("$device.displayName: $cmd endpoint: $endpoint")
+}
+
+def on() {
+	// parent DTH controls endpoint 1
+	def endpointNumber = 1
+	delayBetween([
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinarySet(switchValue: 0xFF)),
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinaryGet())
+	])
+}
+
+def off() {
+	def endpointNumber = 1
+	delayBetween([
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinarySet(switchValue: 0x00)),
+		encap(endpointNumber, zwave.switchBinaryV1.switchBinaryGet())
+	])
+}
+
+// PING is used by Device-Watch in attempt to reach the Device
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	[encap(1, zwave.switchBinaryV1.switchBinaryGet()), encap(2, zwave.switchBinaryV1.switchBinaryGet())]
+}
+
+// sendCommand is called by endpoint 2 child device handler
+def sendCommand(endpointDevice, commands) {
+	def endpointNumber = 2
+	def result
+	if (commands instanceof String) {
+		commands = commands.split(',') as List
+	}
+	result = commands.collect { encap(endpointNumber, it) }
+	sendHubCommand(result, 100)
+}
+
+def encap(endpointNumber, cmd) {
```

Comment:
@LanKuoWei I think I found your issue. Any time you want to manually send security-encapsulated messages, you should verify that the device is securely included using this check:

`	if(zwaveInfo.zw.contains("s")) {`

I believe the error you were seeing was because either the device's secure inclusion had failed or had not completed.

Reply:
I follow this to fix my DTH,
however Err-Code:34-501 still happen,
but this time is different.

Because I use old DTH(no Err-Code) to test,
it's occur have the same error code!?
I can confirm the old DTH is no problem,
in the past, it can be verified at any time,but now have the fail.

I delete all DTH, let device use default setting to inclusion,
and this default inclusion have the same Err-Code:34-501.

I don't know why Err-Code always occur now,
because DTH is no obvious problem,
I already connect SmartThings Support to help me check,

Please wait me update information.

Reply:
Is it possible that there is an issue with the device's firmware? At this point you would probably need to use a zniffer and capture the z-wave traffic.

Reply:
Sure, I already use zniffer to check this,
when DTH run "zwave.multiChannelAssociationV2.multiChannelAssociationSet(groupingIdentifier: 0x01, nodeId: [zwaveHubNodeId])",
SmartThings HUB will wait some second then show Err-Code:34-501.

Reply:
@LanKuoWei Okay. We'll take a look at that, but in the meantime if you want to get unblocked, since your devices are the only ones on this DTH, if you want to change the multiChannelAssociationSet to the packet you have confirmed works, then go ahead.

Reply:
@greens Thanks for your understanding,
this DTH only this question remains, but multiChannelAssociationSet only can set nodeId: [0x01,0x01] to avoid occur Err-Code:34-501,
if you can accept, I will happy for it, or only wait support find to how to fix this issue,
but I thinks this issue is not DTH problem, so I don't want wait to long, please help me.